transcendence-app/src/shared/relationship/relationship.module.ts

```diff
@@ -0,0 +1,10 @@
+import { Module } from '@nestjs/common';
+import { RelationshipService } from './relationship.service';
+import { DbModule } from '../db/db.module';
+
+@Module({
+  imports: [DbModule],
+  providers: [RelationshipService],
+  exports: [RelationshipService],
+})
+export class RelationshipModule {}
```

Comment:
As stated in https://github.com/42AGV/ft_transcendence/pull/248#discussion_r1004798700 I disagree with having yet another module for this. This has no domain logic in our app, it's just a status between users.

I suspect you had this approach due to this being in a new sql table, but I don't see a problem consuming this block repository from users module. The new table is clearly tied to users :crying_cat_face:

If we fear user controller/service etc starts growing we could split controller in some files to split logic.

Reply:
I also don't understand where is this thing coming from. Was it here yesterday when I reviewed it?

Reply:
Ok, I will move this logic into the Users module.

@Damrod, this was the `BlockModule` yesterday. I renamed it to the `RelationshipModule` today to add the friendship relation in the same module.

Reply:
![](https://media.tenor.com/eqLNYv0A9TQAAAAC/swap-indiana-jones.gif)

Reply:
Done :rocket:
I've added this change in #258 because it's the last PR from the stack of PRs related to this feature

Reply:
You may think it's fun, but I've spent one hour tracking down bugs, since I decided to merge your branch into my latest branch, to try and not have merge conflicts.

Reply:
Sorry, @Damrod.

I think squashing the PRs does not help because if we stack PRs and a squash creates a different commit with all the commits into one, the PRs on top lose the reference to the previous commits.
What do we think about changing our workflow to a merge commit even if the git history will become less "clean"?

cc @fizcamposancos @vdedios

Reply:
I agree. No more squashing, as long as we have so many dependencies.

Reply:
I think, rather than a matter of squashing being the problem, I see the problem is indeed working over other's ongoing work when it has not been merged yet.

Not merged code is prone to change and if one decides to start working over other's, on going, work one may be responsible to keep track of updates because the base branch will likely have changes 🤷‍♂️

Not blaming anyone, is the only thing that can be done now, just pointing out the real problem

Reply:
Squash and merge, or rebase and merge, are incompatible with stacked PRs. We have to become git wizards to solve merge conflicts.

Stacking PRs + merge commits are very beneficial and, in my opinion, could work better with the way we work asynchronously.
We can do small PRs, and if one of the previous PRs takes more time to review or merge, we can continue working on other PRs without too many merge conflicts.

I think we are gaining history readability at the cost of velocity.

Reply:
fportela said
> Si te soy sincero tengo claro que es el squash merge o el merge commit. Lo lei en su día pero ya no me acuerdo. Pero en cualquier caso, si sirve para ahorrar problemas, a costa de trazabilidad, por mi no hay fallo 
I wish he had said so in public. Anyhow, I proceed to enable merge commits into master.